Glads gl

RupertGladstone opened the pull request:
Requesting merge of glads_gl into devel

I ran the full ctest on my laptop. Same results for glads_gl and devel. 94% success.

RupertGladstone committed:
Adding routine for calculating GlaDS cross GL flux (total from both
channels and sheet system).

RupertGladstone committed:
Renamed "workvar" to human readable names

RupertGladstone committed:
tidying up "cycelement" decisions

RupertGladstone committed:
Adding new switches (not active yet)...

RupertGladstone committed:
New switches implemented for "calving", and masking code moved to function.

RupertGladstone committed:
Fixed null pointer to solverparams

RupertGladstone committed:
Minor bug fixes re indexing

RupertGladstone committed:
Merge branch 'elmerice' into glads_gl

RupertGladstone committed:
updating grounded melt solver for GlaDS

RupertGladstone committed:
Adding option to not scale As according to effective pressure when converting Weertman to Coulomb coefficients

RupertGladstone committed:
Merge branch 'elmerice' into glads_gl

chekki2mo committed:
reduce output print in GroundedSolver

RupertGladstone committed:
bug fix for nonlinear Budd sliding in SSA

RupertGladstone committed:
Add hybrid (between Schoof/Gag and Joughin versions) regularised coulomb sliding law to SSAMaterials

RupertGladstone committed:
Integrated GMvalid with GroundedSolver.
See new notes at start of GroundedSolver.F90 for usage.

RupertGladstone committed:
Bug fix 'SSA Friction need N' info statement

RupertGladstone committed:
updated info message from GroundedSolver

RupertGladstone committed:
Merge branch 'elmerice' into glads_gl

RupertGladstone committed:
Modifying GlaDS to use specified mask instead of compbined hard coded
grounded mask and GMcheck options.

fgillet committed:
Documentation Covariance Utils

fgillet committed:
documentation

improve documentation CovarianceUtils

fgillet committed:
add header to solver files

fgillet committed:
Add test cases for CovarianceVectorMutliply

fgillet committed:
add test BackgroundErrorCostSolver

fgillet committed:
improve documentation

RupertGladstone committed:
updating GlaDS masking to optionally use the connectivity mask or grounded mask

Morlocke committed:
A selection of updates to get the calving-hydrology coupled model up-and-running again (more to come over the following weeks)

Morlocke committed:
Resolve merge conflict - we already get SolverParams before the FirstTime IF

fgillet committed:
Update Documentation

fgillet committed:
Update documentation

Morlocke committed:
Further changes to make the calving-hydrology coupling work again - mostly edits to my own janky calving code (also some of Joe's!).

RupertGladstone committed:
Add max N to SSA sliding laws

RupertGladstone committed:
Merge branch 'elmerice' into glads_gl

RupertGladstone committed:
tidying glads GL flux calcs

tzwinger committed:
Merge branch 'devel' into elmerice

RupertGladstone committed:
fix a bug in my earlier modifications based on Samuel's "calving" option
(the bug was introduced by my code and occasionally prevented partitions containing
sections of grounding lien from having any channel system)

RupertGladstone committed:
GlaDSCoupledSolver: Bug fix to the looping for averaging the contribution to sheet discharge from different elements.

RupertGladstone committed:
Merge branch 'elmerice' into glads_gl

RupertGladstone committed:
(attempted to) fix calculation of GlaDS sheet contribution to flow across grounding line.

RupertGladstone committed:
Allow both min and max coupled iterations to be specified for GlaDSCoupledSOlver

RupertGladstone committed:
Adding connectivity option for grounded mask to remove isolated grounded regions in shelves

RupertGladstone committed:
updating connectivity mask options to allow for removal of isolated
regions both upstream, and downstream of the GL.

Morlocke committed:
Updates to PlumeSolver to fix a case where a plume actually has a co-ordinate of 0

Morlocke committed:
Now with all that old code taken out!

Morlocke committed:
Minor tweak to GlaDS and PlumeSolver to allow the plumes to use the new GL flux routine

RupertGladstone committed:
Added user option to define mask name for SSA sliding (default still GroundedMask)

RupertGladstone committed:
Additional hybrid RC sliding parameterisation for SSA

tzwinger committed:
Merge branch 'devel' into elmerice

RupertGladstone committed:
Report SSA friction mask name at info level 5

RupertGladstone committed:
Merge branch 'elmerice' into glads_gl

RupertGladstone committed:
Allow SSA code to use a different grounded mask by setting
"Grounded Mask Variable Name" in constants section.
(implemented in SSAmaterial properties, SSA solver and thickness solver;
default is GroundedMask for backward compatibility)

RupertGladstone committed:
Merge branch 'devel' into elmerice

RupertGladstone committed:
Max sheet thickness for GlaDS applied after sheet thickness update.

RupertGladstone committed:
Merge branch 'elmerice' into glads_gl

RupertGladstone committed:
Allow 3D velocity for calculting sliding speed in grounded melt solver

RupertGladstone committed:
tweaking adaptive timestepping info statements

RupertGladstone committed:
reduced info level of adaptivity info from 12 to 7

RupertGladstone committed:
Merge branch 'devel' into elmerice

RupertGladstone committed:
Merge branch 'elmerice' into glads_gl

RupertGladstone committed:
Allow 4D flow solution to be passed to grounded melt solver (ignores 4th dimension, assumed to be pressure)

RupertGladstone committed:
Committing Yu Wang's option to use regularised Coulomb sliding in SSA optimisations

RupertGladstone committed:
Merge branch 'devel' into elmerice

RupertGladstone committed:
Merge branch 'elmerice' into glads_gl

RupertGladstone committed:
resolve minor merge error in GroundedSolver.F90

RupertGladstone committed:
Merge branch 'devel' into glads_gl

RupertGladstone committed:
Debug MeshUtils change re InDofs.

RupertGladstone committed:
Additional optional hydrology-based checks for calculating grounded melt

RupertGladstone committed:
Merge branch 'devel' into glads_gl

RupertGladstone committed:
Merge branch 'devel' into glads_gl

juharu reviewed (commented):
Hi

Haven't  looked at the ElmerIce changes, but a few small comments regarding the elmer solver library changes below.

This change to "MeshUtils.F90": maybe you could search for the space
in the mesh path instead of assuming "."    and hence index 2 ?

  ELSE
       DO s=1,Model % NumberOfSolvers
         !Need to only look at solvers that are going to run on this mesh
         TargetMesh = ListGetString(Model % Solvers(s) % Values, 'Mesh', GotIt)
         DO i=1,6
           DO j=1,10
             inDofs(j,i) = MAX(Indofs(j,i),MAXVAL(Model % Solvers(s) % Def_Dofs(j,:,i)))
             IF(GotIt) THEN
               !This assumes your meshes all start '. '
               IF (LEN_TRIM(Model % Solvers(s) % Mesh % Name) > 0) THEN
                 IF(TRIM(Model % Solvers(s) % Mesh % Name) .NE. TRIM(TargetMesh(2:))) THEN
                   CYCLE
                 END IF
               END IF
             ELSE
               inDofs(j,i) = MAX(Indofs(j,i),MAXVAL(Model % Solvers(s) % Def_Dofs(j,:,i)))
             END IF
           END DO
         END DO

SolverUtils.F90:

     IF ( .NOT. ASSOCIATED( Solver % Matrix ) .OR. &
          .NOT. ASSOCIATED( Solver % Variable % Values ) ) RETURN
          .NOT. ASSOCIATED( Solver % Variable ) ) RETURN

Maybe add
    IF(.NOT.  ASSOCIATED(Solver % Variable Values) ) RETURN
after the changed line ? 

Br, Juha

Morlocke committed:
Update MeshUtils.F90

In response to Juha's comments on Rupert's merge.

Morlocke committed:
Update SolverUtils.F90

In response to Juha's comments on Rupert's merge commit.

raback commented:
The last change in SolverUtils.F90 does not compile.

Morlocke committed:
Update SolverUtils.F90

Deleted the unfortunate space that broke everything

Morlocke committed:
Update SolverUtils.F90

OK, so it's Solver % Values, not VariableValues....

Morlocke committed:
Update SolverUtils.F90

I can't see anything left in the statement at l.9795 that should cause it not to compile?

tzwinger commented on elmerice/Solvers/CalvingGeometry.F90:
Calving3D test fails for me. Output
Backtrace for this error:
#0  0x7fa570f125af in ???
#0  0x7f5710e885af in ???
#1  0x7fa540d92cd3 in __calvinggeometry_MOD_getdomainedge
        at /home/zwinger/Source/Elmer_devel/elmerfem/elmerice/Solvers/CalvingGeometry.F90:2424
#2  0x7fa540ddba97 in frontadvance3d_
        at /home/zwinger/Source/Elmer_devel/elmerfem/elmerice/Solvers/CalvingFrontAdvance3D.F90:197

RupertGladstone replied:
@tzwinger can you confirm that this test passes for you on devel branch? For me, on my laptop, it fails on both devel and glads_gl.

raback commented:
This PR was left in limdo. I am running the test of tentative merge now. If tests pass I guess we can merge? Several people have reviewed the changes, particularly to library.

RupertGladstone committed:
channels hack for the case of a retreating grounding line
(needs a better solution)

RupertGladstone committed:
Adding checks to cmake build process

RupertGladstone committed:
Added code for Grounding Zone Effetive Pressure Threshold (GZEPT) scaling for
basal resistance near the grounding line in Stokes simulations (mainly for
Weertman sliding).

RupertGladstone committed:
Additional comments for EffectivePressure and GZEPT_scaling functions

Morlocke committed:
In preparation for merging in the changes for the all-singing,
all-dancing coupled model

Morlocke committed:
Merge branch 'glads_gl' of https://www.github.com/ElmerCSC/elmerfem into glads_gl
